### PR TITLE
Reduce number of calls from oraclePriceData

### PR DESCRIPTION
**Pull request by sennett**
# [`oraclePriceData$` doesn't return everything by default](https://app.shortcut.com/oazo-apps/story/5173/make-pricedata-great-again)

Reduces infura calls:
- on page load: 329 -> 239
- on every block 289 -> 199

## Changes 👷‍♀️

- `oraclePriceData$` takes a list of required data along with the token and does not request the data it does not need to provide
  
## How to test 🧪
  <Please explain how to test your changes>

- oracles page `/oracles`
- positions overview page - borrow, multiply, earn positions
- all product cards for all positions

### CODE REVIEW

File: blockchain/jsonRpcBatchProvider.ts [hunk 1]
```diff
@@ -58,6 +61,7 @@ export class JsonRpcBatchProvider extends providers.JsonRpcProvider {
           provider: this,
         })
 
+        console.log(`sending batch tot calls ${calls}`)
```
Comment (sennett) [inline blockchain/jsonRpcBatchProvider.ts:64]: I will remove this in a future PR - still using it.

File: blockchain/prices.test.ts [hunk 2]
```diff
@@ -106,3 +109,152 @@ describe('createTokenPriceInUSD$', () => {
...
+        runAssertion: (result) => expect(moment(result.nextPriceUpdate).unix()).eq(1657815532),
+        streamsCalled: ['zzz$', 'hop$'],
+        streamsNotCalled: ['peek$', 'peep$'],
+      })
+    })
+
+    it('calls zzz$ and hop$ for priceUpdateInterval', () => {
+      runTest({
+        requestedValue: 'priceUpdateInterval',
+        runAssertion: (result) => expect(result.priceUpdateInterval?.toString()).eq('3600000'),
```
Comment (BackToTheCode) [inline blockchain/prices.test.ts:245]: This is fantastic

File: helpers/productCards.ts [hunk 3]
```diff
@@ -563,24 +563,24 @@ export function createProductCardsWithBalance$(
...
+            switchMap((tokenPrice) => {
               return of({
                 token: ilk.token,
                 ilk: ilk.ilk as Ilk,
                 liquidationRatio: ilk.liquidationRatio,
                 liquidityAvailable: ilk.ilkDebtAvailable,
                 stabilityFee: ilk.stabilityFee,
                 debtFloor: ilk.debtFloor,
-                currentCollateralPrice: priceInfo.currentCollateralPrice,
+                currentCollateralPrice: tokenPrice.currentPrice,
```
Comment (BackToTheCode) [inline helpers/productCards.ts:583]: so much better

**Review by BackToTheCode — APPROVED** [review]
Really great - nothing to add. Assume you're aware of the failing tests.

File: blockchain/prices.ts [hunk 4]
```diff
@@ -213,53 +213,111 @@ export function calculatePricePercentageChange(current: BigNumber, next: BigNumb
...
 export function createOraclePriceData$(
   context$: Observable<Context>,
   pipPeek$: (token: string) => Observable<[string, boolean]>,
   pipPeep$: (token: string) => Observable<[string, boolean]>,
   pipZzz$: (token: string) => Observable<BigNumber>,
   pipHop$: (token: string) => Observable<BigNumber>,
-  token: string,
-): Observable<OraclePriceData> {
+  { token, requestedData }: OraclePriceDataArgs,
+): Observable<Partial<OraclePriceData>> {
```
Comment (paszkowskiDamian) [inline blockchain/prices.ts:228]: by Partial you mean that `token` is also optional?

Comment (sennett) [reply blockchain/prices.ts:228]: No - the `Partial` wraps the return value from `createOraclePriceData$`, not the arguments to `createOraclePriceData$`.

**Pull request merged**